### ISSUE TITLE
Lambda constructors added in version 5.2.23 breaks StructureMap support

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus#4955]
Reported by a user via Gitter - https://gitter.im/Particular/NServiceBus?at=59bbaa71c101bc4e3acd1912
Relevant code change: https://github.com/Particular/NServiceBus/pull/4899/files#r139117439

@Particular/nservicebus-maintainers since this blocks StructureMap users I propose we treat this as a patch?

**Comment by andreasohlund** [comment]
Probably breaks other containers as well, wonder if making the ctor internal is enough?

**Comment by andreasohlund** [comment]
I've confirmed the issue with a minimal repro

```
 class Program
    {
        static void Main(string[] args)
        {
            var ec = new BusConfiguration();

            ec.UseContainer<StructureMapBuilder>();
            ec.UsePersistence<InMemoryPersistence>();

            Bus.Create(ec).Start();

            Console.ReadKey();
        }
    }

    class Error: IProvideConfiguration<MessageForwardingInCaseOfFaultConfig>
    {
        public MessageForwardingInCaseOfFaultConfig GetConfiguration()
        {
            return new MessageForwardingInCaseOfFaultConfig
            {
                ErrorQueue ="error"
            };
        }
    }
```

This prevents the endpoint from running since the timeout poller trips the circuit breaker with

```
2017-09-15 15:04:09.330 WARN  NServiceBus.Timeout.Hosting.Windows.TimeoutPersisterReceiver Failed to fetch timeouts from the timeout storage
StructureMap.StructureMapBuildPlanException: Unable to create a build plan for concrete type System.DateTime

new DateTime(Int64 ticks)
  ? Int64 ticks = Required primitive dependency is not explicitly defined


1.) Attempting to create a BuildPlan for Instance of System.DateTime -- System.DateTime
2.) Container.GetInstance<System.DateTime>()

```

**Comment by andreasohlund** [comment]
I've confirmed that removing the lambda ctors fixes the issue, will open a PR

**Comment by andreasohlund** [comment]
https://github.com/Particular/NServiceBus/pull/4958

**Comment by andreasohlund** [comment]
Verified that v6.X isn't effected since we register the affected types using lambda instead of letting the container satisfy the deps

**Comment by bording** [comment]
Fixed by #4958
